Framework/DataHandling/src/LoadILLDiffraction.cpp

```diff
@@ -513,7 +518,10 @@ void LoadILLDiffraction::fillStaticInstrumentScan(const NXUInt &data,
     auto &spectrum = m_outWorkspace->mutableY(i);
     auto &errors = m_outWorkspace->mutableE(i);
     const auto tubeNumber = (i - NUMBER_MONITORS) / m_sizeDim2;
-    const auto pixelInTubeNumber = (i - NUMBER_MONITORS) % m_sizeDim2;
+    auto pixelInTubeNumber = (i - NUMBER_MONITORS) % m_sizeDim2;
+    if (m_instName == "D2B" && tubeNumber % 2 == 1) {
+      pixelInTubeNumber = D2B_NUMBER_PIXELS_IN_TUBES - 1 - pixelInTubeNumber;
+    }
```

Comment:
Can this be done without giving up the const-ness of `pixelInTubeNumber`? Perhaps by having another variable such as `pixelInFirstTubeNumber`.

Reply:
It can be done probably with a lambda function, but I don't think this or introducing a new variable will help the clarity. The `pixelInTubeNumber` is immediately used in the next line, and evidently as an input, so I don't think giving up the const-ness is a big sacrifice here.